poopcam.ts

```diff
@@ -28,14 +29,25 @@ export abstract class PoopCam {
         }
 
         if (cammer === undefined) {
-            cammer = {userName: userName, requestCount: 0};
+            cammer = {userName: userName, requestCount: 0, date: new Date(0)};
 
         }
-
-        cammer.requestCount++;
+        let date: Date = new Date();
+        let rate_limit = false;
+        if (cammer.date === undefined) {
+            rate_limit = true;
+        } else {
+            rate_limit = (cammer.date.getTime() + 60000 < date.getTime()); //rate limit of 1 min
+        }
+        if (rate_limit) { 
+            cammer.requestCount++;
+        }
+        cammer.date = date;
```

Comment:
We should only to this if we are not limiting them, otherwise someone who happens to !poopcam a little more frequently than  every 1 minute will instead never be able to increment their count.

Reply:
that is on purpose to make requests less frequent and to punish spamming

Reply:
Should we add something that lets them know then? Otherwise someone who is not spamming but doesn't know the rate limit would never get a poopcam

Reply:
Meh this isn't that big of a deal actually.